numpy/core/src/common/ufunc_override.c

```diff
@@ -93,8 +93,19 @@ PyUFuncOverride_GetOutObjects(PyObject *kwds, PyObject **out_kwd_obj, PyObject *
         return 0;
     }
     if (PyTuple_CheckExact(*out_kwd_obj)) {
-        *out_objs = PySequence_Fast_ITEMS(*out_kwd_obj);
-        return PySequence_Fast_GET_SIZE(*out_kwd_obj);
+        /*
+         * The C-API recommends calling PySequence_Fast before any of the other
+         * PySequence_Fast* functions. This is required for PyPy
+         */
+        PyObject *seq = PySequence_Fast(*out_kwd_obj, "Could not convert object to sequence");
+        int ret;
+        if (seq == NULL) {
+            return -1;
+        }
+        *out_objs = PySequence_Fast_ITEMS(seq);
+        ret = PySequence_Fast_GET_SIZE(seq);
+        Py_SETREF(*out_kwd_obj, seq);
```

Comment:
The comment above says that `*out_kwd_obj` is a borrowed reference, but as a result of `PySequence_Fast`, it's now a new reference, so the caller needs to decref it.

Reply:
Similarly, `Py_SETREF` is incorrect here - it's not safe to decref `*out_kwd_obj`, as you don't own it. Simply
```
*out_kwd_obj = seq
```
is what you want here

Reply:
Good catch. But now `seq` may not be a borrowed ref. I changed the interface so that `*out_kwd_obj` is never a borrowed ref.

Reply:
The fact that tests pass indicates there may be a leak of kwds['out'] somewhere.

Reply:
Tests pass because you were swapping a refcount between seq and the original out_kwd_obj. In CPython, those objects happen to be the same, so you got away with it.